app/controllers/uploaded_files_controller.rb

```diff
@@ -48,4 +60,14 @@ def show
     def uploaded_file_params
       params.require(:uploaded_file).permit(:attachment, :lecture)
     end
+
+    def validate_destroy_rights
+      @uploaded_file = UploadedFile.find(params[:id])
+      @owner = current_user == @uploaded_file.author
+      @course_file_and_course_owner = (@uploaded_file.allowsUpload.class == Course) && (@uploaded_file.allowsUpload.creator_id == current_user.id)
+      @lecture_file_and_lecture_owner = (@uploaded_file.allowsUpload.class == Lecture) && (@uploaded_file.allowsUpload.lecturer_id == current_user.id)
```

Comment:
I think you can omit the `@` in front of the variables. The `@` is only required for class wide variables.
https://www.techotopia.com/index.php/Ruby_Variable_Scope

# Please wait with merge unless this is fixed.

Everything else looks great now and can be merged after this small fix